Client/src/Components/ThemeSwitch/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from "react";
+import { IconButton } from "@mui/material";
+import SunAndMoonIcon from "./SunAndMoonIcon";
+import "./index.css";
+import { useDispatch, useSelector } from "react-redux";
+import { setMode } from "../../Features/UI/uiSlice";
+
+const ThemeSwitch = () => {
+	const mode = useSelector((state) => state.ui.mode);
+	const dispatch = useDispatch();
+
+	const toggleTheme = () => {
+		dispatch(setMode(mode === "light" ? "dark" : "light"));
+	};
+
+	useEffect(() => {
+		document.body.setAttribute("data-theme", mode);
+	}, [mode]);
```

Comment:
I understand the logic from this implementation relies on data-attributes. But we already have a theme on redux. I suggest adapting that to rely on our theme as well. This way you can ditch this useEffect

Reply:
Got it. Let me update the component.